zebra-consensus/src/block.rs

```diff
@@ -116,7 +123,7 @@ where
             let now = Utc::now();
             node_time_check(block.header.time, now)?;
             block.header.is_equihash_solution_valid()?;
-            coinbase_check(block.as_ref())?;
+            coinbase_is_first_check(block.clone())?;
```

Comment:
I don't understand why its preferable to clone an arc here over passing by ref

Reply:
Reverted in 19fd8bcfb42e0366863b5aa1393d49a58f80af81